### ISSUE TITLE
energy consumption

### ISSUE THREAD

**Issue by daydayup999** [ayltai/Android-Lib-VerticalMarqueeTextView#6]
CPU occupancy will be high

private void animateTextView() {
        final Runnable runnable = new VerticalMarqueeTextView.MarqueeRunnable(this.textView);

```
    long previousMillis = 0;

    while (VerticalMarqueeTextView.this.marqueeStarted && !VerticalMarqueeTextView.this.marqueePaused) {
        final long currentMillis = System.currentTimeMillis();

        if (currentMillis >= previousMillis) {
            VerticalMarqueeTextView.this.handler.post(runnable);

            previousMillis = currentMillis + (long)(1000d / VerticalMarqueeTextView.this.marqueeSpeed);
        }
    }

    this.isAnimating = false;
}
```

**Comment by ayltai** [comment]
Right. A better approach should be used.

**Comment by ayltai** [comment]
This issue is fixed in version 1.0.2
